### PR TITLE
Bugfix/334 incorrect sub account display

### LINKED ISSUE #334: Bug: Account Teaser not displaying sub accounts correctly
Sub accounts should display the parent account name as well as the sub-identity name as shown here: 

Screenshot 2021-08-30 at 9 42 58 AM

Currently the teaser is only displaying the parent account name:

Screenshot 2021-08-30 at 9 41 03 AM

This is a regression bug, it was introduced in this [PR](https://github.com/litentry/litentry-app/pull/237)

This is what the old hook (`useAccountDetail`) was returning:

```
const displayFull = subAccountDisplay ? `${display}/${subAccountDisplay}` : display;
```

### PR DESCRIPTION
**Pull request by ahoseinian**
fixes #334
- use `api.derive` to get identity info
- use [polkadot wiki](https://wiki.polkadot.network/docs/learn-identity#sub-accounts) way of showing parent/sub accounts
- make sure it works everywhere

----
<p align="center">
  
</p>

### CODE REVIEW

File: src/api/queryFunctions/getAccountIdentityInfo.ts [hunk 1]
```diff
@@ -1,59 +1,44 @@
...
 
 export async function getAccountIdentityInfo(api: ApiPromise, accountId: string): Promise<IdentityInfo> {
-  const registrationOption = await api.query.identity.identityOf(accountId);
-  const registration = registrationOption.unwrapOr(undefined);
+  const info = await api.derive.accounts.info(accountId);
+
+  if (info) {
+    let display = info.identity.display ?? accountId;
+    if (info.identity.displayParent) {
+      if (info.identity.display) {
```
Comment (silva-fj) [inline src/api/queryFunctions/getAccountIdentityInfo.ts]: NIT:
``` 
display = `${info.identity.displayParent}/${info.identity.display || info.identity.displayParent}`;
```
or 

```
if (info) {
    const display = info.identity.displayParent ? `${info.identity.displayParent}/${info.identity.display || info.identity.displayParent}` : info.identity.display ?? accountId;
    
    return ...
}

**Review by silva-fj — DISMISSED** [review]
Awesome, way cleaner 👍🏼

**Comment by silva-fj** [comment]
In the polkadotjs app the sub accounts also have a green icon like the identities with judgement but the symbol is a link, did you check if we can also get that?

https://user-images.githubusercontent.com/8374946/131304408-a83ac720-a53c-4eed-8442-1b50c3c8bef1.png

**Comment by ahoseinian** [comment]
> In the polkadotjs app the sub accounts also have a green icon like the identities with judgement but the symbol is a link, did you check if we can also get that?
> 
> https://user-images.githubusercontent.com/8374946/131304408-a83ac720-a53c-4eed-8442-1b50c3c8bef1.png

I just checked it, it seems they show the link when the identity has a parent otherwise the check or minus icons.
https://github.com/polkadot-js/apps/blob/bc3e6a50b1507abcabc12321ed798aa86c492c61/packages/react-components/src/AccountName.tsx#L125

I will try to do the same

File: src/api/queryFunctions/getAccountIdentityInfo.ts (hunk 1 above)
Comment (ahoseinian) [reply src/api/queryFunctions/getAccountIdentityInfo.ts]: done ✅

**Comment by ahoseinian** [comment]
It exploded a bit while adding the link icon. I did some refactoring but it mostly reduced the code so should be all fine.

File: src/api/hooks/useAccountIdentityInfo.ts [hunk 2]
```diff
@@ -2,13 +2,11 @@ import {ApiPromise} from '@polkadot/api';
...
+export function useAccountIdentityInfo(accountId: string) {
   return useApiQuery(
     ['account_identity', accountId],
     (api: ApiPromise) => {
-      if (accountId) {
-        return getAccountIdentityInfo(api, accountId);
-      }
+      return getAccountIdentityInfo(api, accountId);
     },
     {staleTime: 1000 * 60, enabled: Boolean(accountId)},
```
Comment (silva-fj) [inline src/api/hooks/useAccountIdentityInfo.ts]:
```suggestion
    {staleTime: 1000 * 60},
```

**Review by silva-fj — DISMISSED** [review]

**Review by silva-fj — APPROVED** [review]

**Pull request merged**